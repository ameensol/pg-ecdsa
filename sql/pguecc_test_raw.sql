create extension pgcrypto;
create extension pguecc;

--
-- ecdsa_sign_raw
---

-- invalid curve
select ecdsa_sign_raw(E'\\x1234', E'\\x1234', 'baz');

-- invalid key size
select ecdsa_sign_raw(E'\\x1234', E'\\x1234', 'secp256k1');

-- valid key size (output is non-deterministic so just check the length)
select length(ecdsa_sign_raw(E'\\x000000000000000000000000000000000000000000', E'\\x1234', 'secp160r1'));

-- null values
select ecdsa_sign_raw(NULL, '1234', 'secp160r1');
select ecdsa_sign_raw(E'\\x00', NULL, 'secp160r1');
select ecdsa_sign_raw(E'\\x00', '1234', NULL);

--
-- ecdsa_verify_raw
---

-- invalid curve
select ecdsa_verify_raw(E'\\x1234', E'\\x1234', E'\\x1234', 'baz');

-- invalid pubkey
select ecdsa_verify_raw(E'\\x1234', E'\\x1234', E'\\x1234', 'secp256k1');

-- valid signatures (keys generated with OpenSSL)
select ecdsa_verify_raw(
    E'\\xDCB01EA59807D6B8F07E685A9AF369D70554586EF982B35AB2087F3DEABE5065D5130A53175377EB',
    E'\x1234',
    ecdsa_sign_raw(E'\\x0001997CBE1B9E90DAB23322F557E0A2510DA101FA', E'\x1234', 'secp160r1'),
    'secp160r1'
);
