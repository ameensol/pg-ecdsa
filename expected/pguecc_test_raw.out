create extension pgcrypto;
create extension pguecc;
--
-- ecdsa_sign_raw
---
-- invalid curve
select ecdsa_sign_raw(E'\\x1234', E'\\x1234', 'baz');
ERROR:  Invalid curve: baz
-- invalid key size
select ecdsa_sign_raw(E'\\x1234', E'\\x1234', 'secp256k1');
ERROR:  Invalid private key size for curve secp256k1: 2 (should be 32)
-- valid key size (output is non-deterministic so just check the length)
select length(ecdsa_sign_raw(E'\\x000000000000000000000000000000000000000000', E'\\x1234', 'secp160r1'));
 length 
--------
     40
(1 row)

-- null values
select ecdsa_sign_raw(NULL, '1234', 'secp160r1');
 ecdsa_sign_raw 
----------------
 
(1 row)

select ecdsa_sign_raw(E'\\x00', NULL, 'secp160r1');
 ecdsa_sign_raw 
----------------
 
(1 row)

select ecdsa_sign_raw(E'\\x00', '1234', NULL);
 ecdsa_sign_raw 
----------------
 
(1 row)

--
-- ecdsa_verify_raw
---
-- invalid curve
select ecdsa_verify_raw(E'\\x1234', E'\\x1234', E'\\x1234', 'baz');
ERROR:  Invalid curve: baz
-- invalid pubkey
select ecdsa_verify_raw(E'\\x1234', E'\\x1234', E'\\x1234', 'secp256k1');
ERROR:  Invalid public key size for curve secp256k1: 2 (should be 64)
-- valid signatures (keys generated with OpenSSL)
select ecdsa_verify_raw(
    E'\\xDCB01EA59807D6B8F07E685A9AF369D70554586EF982B35AB2087F3DEABE5065D5130A53175377EB',
    E'\x1234',
    ecdsa_sign_raw(E'\\x0001997CBE1B9E90DAB23322F557E0A2510DA101FA', E'\x1234', 'secp160r1'),
    'secp160r1'
);
 ecdsa_verify_raw 
------------------
 t
(1 row)

